tests/js/transformationsTests.js

```diff
@@ -0,0 +1,138 @@
+        [{value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}], // total === "string"
+        [{value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: 100}, {value: null}, {value: 0}, {value: null}, {value: 2272.7272727272725}, {value: 4545.454545454545}], // total === 2.2
+        [{value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}], // total === "2.2"
+        [{value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: 0}, {value: null}, {value: 0}, {value: null}, {value: 0}, {value: 0}], // total === 0
+        [{value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}], // total === "0"
+        [{value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: 4.4}, {value: null}, {value: 0}, {value: null}, {value: 100}, {value: 200}], // total === 50
+        [{value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: null}, {value: 2.2}, {value: null}, {value: 0}, {value: null}, {value: 50}, {value: 100}] // total === 100
+    ];
+
+    jqUnit.test("Test gpii.chartAuthoring.transforms.percentage", function () {
+        fluid.each(gpii.tests.chartAuthoring.transforms.percentageInputs, function (total, totalIdx) {
+            fluid.each(gpii.tests.chartAuthoring.transforms.percentageInputs, function (value, valIdx) {
+                var transform = {
+                    transform: {
+                        type: "gpii.chartAuthoring.transforms.percentage",
+                        outputPath: "value",
+                        value: value,
+                        total: total
+                    }
+                };
+                var expected = gpii.tests.chartAuthoring.transforms.percentageOutputs[totalIdx][valIdx];
+                gpii.tests.chartAuthoring.transforms.assertTransformation("The value: '" + value + "' and total: '" + total + "' should have been transformed", {}, transform, expected);
+            });
+        });
+    });
+
+    gpii.tests.chartAuthoring.transforms.reduceAdd = function (value, currentValue) {
```

Comment:
The extraction of the values from an object is just adding confusion to what is essentially the same algorithm. I suggest that rather than building the extraction into the accumulate algorithm, you instead run a fluid.transform (or fluid.getMembers) operation on the initial array to put it into the form expected by the straightforward algorithm written here (reduceAdd)